freqtrade/freqai/__init__.py

```diff
@@ -0,0 +1,14 @@
+# ensure users can still use a non-torch freqai version
+try:
+    from freqtrade.freqai.tensorboard import TensorBoardCallback, TensorboardLogger
+    TBLogger = TensorboardLogger
+    TBCallback = TensorBoardCallback
+except ModuleNotFoundError:
+    from freqtrade.freqai.tensorboard import BaseTensorBoardCallback, BaseTensorboardLogger
+    TBLogger = BaseTensorboardLogger  # type: ignore
+    TBCallback = BaseTensorBoardCallback  # type: ignore
```

Comment:
i **think** i understand what you're trying to do here : 
the initial import should fail if torch is not installed - and fall back to the `BaseTensorboardLogger`.

That's however not what's happening - this import will always work (unless xgboost is not installed, in which case both try and except blocks will fail as they both import from the same location).

Instead, you'll hit a `ModuleNotFound` exception once TensorboardLogger is initialized - as the import of torch is only happening once you're creating an instance - but at that point, there's no "graceful fallback"  - so it'll simply fail.


What you'll want to do instead is to split the 2 logics  into 2 files (`Base*` in one file, "actual" in the other) - and move the torch import to the top of the 2nd file (so it's failing on import).

Then this logic will work / be effective, and the fallback to "base" will actually work (the imports will then come from 2 separate files).

---

on top of that, i'd suggest to move this whole logic (callbacks + the content of this file) into it's own package (directory) - maybe called `tensorboard/...` - so it's clear that these 3 files belong together.

It'll obviously change the imports to `from freqtrade.freqai.tensorboard import xxx` ... but i think that's better than having it top-level - where it might cause problems with circular imports (you're importing from the __init__.py on files within that package - which can in some cases lead to problems).